packages/@ember/-internals/glimmer/lib/helpers/unique-id.ts

```diff
@@ -41,7 +41,7 @@ export default internalHelper((): Reference<string> => {
 // implementation details into an intimate API. It also ensures that the UUID
 // always starts with a letter, to avoid creating invalid IDs with a numeric
 // digit at the start.
-function uniqueId() {
+export function uniqueId() {
```

Comment:
I am unsure about the defaulted export of this file, which returns a type `HelperDefinitionState`, so I also exported the actual function here.

Reply:
I *believe* this is correct, but we actually don't have *literally any* prior art here as far as I can tell—this is the first time we've really needed to think about helpers-which-are-functions-and-should-be-usable-in-JS! Accordingly, I'm going to ask @wycats (who implemented) or @rwjblue or @chancancode (who have more context than I do) to confirm. I *believe* we're doing the `internalHelper()` dance here with the default export to make sure that this works as a helper when invoked as `unique-id`. However, that export will, as far as I can tell, *not* be usable in JS directly, but *will* be usable in a template.

But I think we *do* want `uniqueId` to be usable in JS contexts exactly like in HBS contexts, and therefore I think what you've done here is correct.

Reply:
All right, we chatted about this today and we on the Framework Core Team side need to do some follow-up work (to which you will be welcome to contribute if you like!) for simple function-based helpers like this one and how they relate to the ability to use [plain old functions as helpers](https://blog.emberjs.com/plain-old-functions-as-helpers) in Ember. For now, however, we are going to make this work *only* as a traditional helper, rather than as a plain JS function.

Accordingly, we can drop this import and make a couple other tweaks, and this should be good to go!

```suggestion
function uniqueId() {
```

Reply:
All right, we chatted about this today and we on the Framework Core Team side need to do some follow-up work (to which you will be welcome to contribute if you like!) for simple function-based helpers like this one and how they relate to the ability to use [plain old functions as helpers](https://blog.emberjs.com/plain-old-functions-as-helpers) in Ember. For now, however, we are going to make this work *only* as a traditional helper, rather than as a plain JS function.

Accordingly, we can drop this import and make a couple other tweaks, and this should be good to go!

```suggestion
function uniqueId() {
```

Reply:
makes sense to me! pls loop me in on the follow-up work!